src/controllers/plugins/LuaUtilities.hpp

```diff
@@ -2,44 +2,30 @@
 
 #ifdef CHATTERINO_HAVE_PLUGINS
 
-#    include "common/QLogging.hpp"
-
-extern "C" {
 #    include <lua.h>
 #    include <lualib.h>
-}
 #    include <magic_enum/magic_enum.hpp>
 #    include <QList>
+#    include <sol/state_view.hpp>
 
 #    include <cassert>
-#    include <optional>
 #    include <string>
 #    include <string_view>
 #    include <type_traits>
-#    include <variant>
-#    include <vector>
 struct lua_State;
-class QJsonObject;
-namespace chatterino {
-struct CommandContext;
-}  // namespace chatterino
 
 namespace chatterino::lua {
 
-namespace api {
-    struct CompletionList;
-    struct CompletionEvent;
-}  // namespace api
-
-constexpr int ERROR_BAD_PEEK = LUA_OK - 1;
-
 /**
  * @brief Dumps the Lua stack into qCDebug(chatterinoLua)
  *
  * @param tag is a string to let you know which dump is which when browsing logs
  */
 void stackDump(lua_State *L, const QString &tag);
 
+// This is for calling stackDump out of gdb as it's not easy to create a QString there
+const QString GDB_DUMMY = "GDB_DUMMY";
```

Comment:
Shouldn't this be in the source file?

Reply:
It needs to be in a header so it's accessible. It could have a default value assigned in a source file although that might have linking issues.

Reply:
Could this be a null-string? That would be constexpr + all null. I guess we can't have it be `inline` because we don't use it.

One should be able to debug the lua stack without `stackDump` though. For GDB there's https://github.com/mkottman/lua-gdb-helper (might be a bit outdated). And for VS there's https://github.com/huoyang11/lua_debug_natvis (that one is a bit outdated and incompatible with qt, but not too hard to fix). There's probably something for LLDB as well.